source/common/http/header_map_impl.h

```diff
@@ -16,38 +16,33 @@ namespace Http {
 
 /**
  * These are definitions of all of the inline header access functions described inside header_map.h
- *
- * When a non-const reference or pointer to a HeaderEntry is returned, the internal byte size count
- * will be cleared, since HeaderMap will no longer be able to accurately update the size of that
- * HeaderEntry.
- * TODO(asraa): Remove functions with a non-const HeaderEntry return value.
  */
 #define DEFINE_INLINE_HEADER_FUNCS(name)                                                           \
 public:                                                                                            \
   const HeaderEntry* name() const override { return inline_headers_.name##_; }                     \
-  HeaderEntry* name() override {                                                                   \
-    cached_byte_size_.reset();                                                                     \
-    return inline_headers_.name##_;                                                                \
-  }                                                                                                \
-  HeaderEntry& insert##name() override {                                                           \
-    cached_byte_size_.reset();                                                                     \
-    return maybeCreateInline(&inline_headers_.name##_, Headers::get().name);                       \
+  void append##name(absl::string_view data, absl::string_view delimiter) override {                \
+    HeaderEntry& entry = maybeCreateInline(&inline_headers_.name##_, Headers::get().name);         \
+    addSize(HeaderMapImpl::appendToHeader(entry.value(), data, delimiter));                        \
```

Comment:
Add TODO here to make these macros expand into single virtual calls, adding new pure methods.